src/telempostdaemon.c

```diff
@@ -0,0 +1,755 @@
+
+        /* Zero out expired records for record array */
+        for (int i = current_minute + 1; i < (blank_slots + current_minute + 1); i++) {
+                array[i % TM_RATE_LIMIT_SLOTS] = 0;
+        }
+}
+
+static void set_pollfd(TelemPostDaemon *daemon, int fd, enum fdindex i, short events)
+{
+        assert(daemon);
+        assert(fd);
+
+        daemon->pollfds[i].fd = fd;
+        daemon->pollfds[i].events = events;
+        daemon->pollfds[i].revents = 0;
+}
+
+static void initialize_signals(TelemPostDaemon *daemon)
+{
+        int sigfd;
+        sigset_t mask;
+
+        sigemptyset(&mask);
+
+        if (sigaddset(&mask, SIGHUP) != 0) {
+                telem_perror("Error adding signal SIGHUP to mask");
+                exit(EXIT_FAILURE);
+        }
+
+        if (sigaddset(&mask, SIGINT) != 0) {
+                telem_perror("Error adding signal SIGINT to mask");
+                exit(EXIT_FAILURE);
+        }
+
+        if (sigaddset(&mask, SIGTERM) != 0) {
+                telem_perror("Error adding signal SIGTERM to mask");
+                exit(EXIT_FAILURE);
+        }
+
+        if (sigaddset(&mask, SIGPIPE) != 0) {
+                telem_perror("Error adding signal SIGPIPE to mask");
+                exit(EXIT_FAILURE);
+        }
+
+        if (sigprocmask(SIG_BLOCK, &mask, NULL) == -1) {
+                telem_perror("Error changing signal mask with SIG_BLOCK");
+                exit(EXIT_FAILURE);
+        }
+
+        sigfd = signalfd(-1, &mask, 0);
+        if (sigfd == -1) {
+                telem_perror("Error creating the signalfd");
+                exit(EXIT_FAILURE);
+        }
+
+        daemon->sfd = sigfd;
+
+        set_pollfd(daemon, sigfd, signlfd, POLLIN);
+}
+
+static void initialize_rate_limit(TelemPostDaemon *daemon)
+{
+        for (int i = 0; i < TM_RATE_LIMIT_SLOTS; i++) {
+                daemon->record_burst_array[i] = 0;
+                daemon->byte_burst_array[i] = 0;
+        }
+        daemon->rate_limit_enabled = rate_limit_enabled_config();
+        daemon->record_burst_limit = record_burst_limit_config();
+        daemon->record_window_length = record_window_length_config();
+        daemon->byte_burst_limit = byte_burst_limit_config();
+        daemon->byte_window_length = byte_window_length_config();
+        daemon->rate_limit_strategy = rate_limit_strategy_config();
+}
+
+static void initialize_record_delivery(TelemPostDaemon *daemon)
+{
+        daemon->record_retention_enabled = record_retention_enabled_config();
+        daemon->record_server_delivery_enabled = record_server_delivery_enabled_config();
+}
+
+void initialize_daemon(TelemPostDaemon *daemon)
+{
+        assert(daemon);
+
+        daemon->bypass_http_post_ts = 0;
+        daemon->is_spool_valid = is_spool_valid();
+        daemon->record_journal = open_journal(JOURNAL_PATH);
+        daemon->fd = inotify_init();
+        if (daemon->fd < 0) {
+                perror("Error initializing inotify");
```

Comment:
This might be a fatal error... Should telempostd exit in this case?

Reply:
True